### DISCUSSION TITLE
VRM AC loads not considering battery power

### DISCUSSION THREAD

**Discussion by mhin76** [mr-manuel/venus-os_dbus-mqtt-battery#24] [Q&A]
Hi Manuel

I do have another question. Should not the battery power/flow be considered to calculate the AC loads?

I would expect AC Loads = PV Inverter + Grid - Battery Power. In the printscreen below the AC Loads should be 1145, because the battery is providing additional 84 Watt. Same also when I charge the battery, the power going into battery should be taken off the AC loads.

![image](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery/assets/49210075/9cc947a2-d521-44a9-8507-364bc856528a)

**Comment by mr-manuel** [comment]
Do you have a MutliPlus or Quattro? If not then you need to install the [emulator](https://github.com/mr-manuel/venus-os_dbus-multiplus-emulator).

**Reply by mhin76** [reply]
Hi Manuel. I have installed the emulator. And also linked the bat monitoring. Bat shows correctly. I noticed that discharge also shows on the emulator discharging. On Charging it still shows bulk. and also here does not calculate correctly:
![image](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery/assets/49210075/e5d09355-5c51-4370-a4bd-d17df68362b2)

and I guess bc it shows bulk on the MultiPlus emulator and not decharging it also does not calculate anything in the summary part:
![image](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery/assets/49210075/3821dd4c-1852-4db0-bc48-b51ea2ed8db9)


on emulator I set the phase in the config to L4, with any other phase it did not calculate correctly the AC loads. not sure if that has any side affect. let me further test that.

hhmmm... where do I need to set on which phase the battery will be charged?

this is what I send to the battery service:
```
var message = {
    'Dc': {
        'Power': global.get("batW"),
        'Voltage': flow.get("batVoltage"),
        'Current': flow.get("batA"),
        'Temperature': flow.get("temperature")
    },
    "InstalledCapacity": 3.2 * 8 * 230,
    'Capacity': 3.2 * 8 * flow.get("batAh"),
    'Soc': flow.get("soc"),
    'System': {
        'MinCellVoltage': flow.get("minCellV"),
        'MaxCellVoltage': flow.get("maxCellV"),
        'NrOfModulesOnline': 8
        
    },
    "Voltages": {
        'Cell1': flow.get("Cell1V"),
        'Cell2': flow.get("Cell2V"),
        'Cell3': flow.get("Cell3V"),
        'Cell4': flow.get("Cell4V"),
        'Cell5': flow.get("Cell5V"),
        'Cell6': flow.get("Cell6V"),
        'Cell7': flow.get("Cell7V"),
        'Cell8': flow.get("Cell8V"),
    }
};
```

**Reply by mhin76** [reply]
I see now why Phase L1 at multiplus emulator gives wrong figures. It recognizes my powermeter via USB as well via the multiplus emulator. and I have added it directly in venus OS. so e.g. phase L1 is counted twice. need to find out how to exclude it on one or the other service.
let me open up a discussion thread on the emulation github respository. it might be related and this behaviour here is a side effect of the other not being setup correctly.
